### PR TITLE
Append list of wrapped `finders` in `sys.meta_path` with list of `non-wrapped` finders

### PR DESCRIPTION
**Pull request by vtomole**
Readthedocs hasn't been able to build on libraries that import Cirq since https://github.com/quantumlib/Cirq/commit/fd257e6c228a0b4c8839ad6ac3d30467476f195e#r55422405.

This is because Sphinx inserts a `MockFinder` to [sys.meta_path](https://github.com/sphinx-doc/sphinx/blob/f127a2ff5d6d86918a5d3ac975e8ab8a24c407d1/sphinx/ext/autodoc/mock.py#L148) but when Sphinx tries to remove the inserted `MockFinder` from said [path](https://github.com/sphinx-doc/sphinx/blob/f127a2ff5d6d86918a5d3ac975e8ab8a24c407d1/sphinx/ext/autodoc/mock.py#L151), the specific finder that Sphinx is looking for doesn't exist because Cirq wrapped it: https://github.com/quantumlib/Cirq/blob/58cda7a4f9718ef0bebd91950af192df8d47b0d6/cirq-core/cirq/_compat.py#L624

This change fixes the issue, but I don't know what consequences it has for other stuff. I'd be glad to hear better suggestions.

If this looks good, we could do the same thing for  [def wrap_module_executions](https://github.com/quantumlib/Cirq/blob/58cda7a4f9718ef0bebd91950af192df8d47b0d6/cirq-core/cirq/_import.py#L138) as well.

### CODE REVIEW

File: cirq-core/cirq/_compat.py [hunk 1]
```diff
@@ -621,6 +621,13 @@ def deprecated_submodule(
             )
 
     def wrap(finder: Any) -> Any:
+        # Sphinx looks for non-wrapped Mockfinders
```
Comment (cduck) [inline cirq-core/cirq/_compat.py:624]: Nit.
```suggestion
        # Sphinx looks for non-wrapped MockFinders
```

File: cirq-core/cirq/_compat_test.py [hunk 2]
```diff
@@ -853,3 +855,67 @@ def _dir_is_still_valid_inner():
...
+    sys.meta_path.insert(0, fake_mockfinder)
+    deprecated_submodule(
+        new_module_name='sphinx_1',
+        old_parent='sphinx_2',
+        old_child='old_ch',
+        deadline='v1.2',
+        create_attribute=False,
+    )
+    # Cleanup sys.metapath after test
+    sys.meta_path.remove(fake_mockfinder)
```
Comment (cduck) [inline cirq-core/cirq/_compat_test.py:921]: Make it clear that this line is what tests the desired behavior.  Or separately `assert fake_mockfinder in sys.meta_path`.

File: dev_tools/conf/mypy.ini [hunk 3]
```diff
@@ -27,4 +27,8 @@ follow_imports_for_stubs = true
 
 # ruamel is a downstream dependency of cirq-rigetti through pyquil.
 [mypy-ruamel.*]
-ignore_missing_imports = true
\ No newline at end of file
+ignore_missing_imports = true
+
+# Mocked modules
```
Comment (cduck) [inline dev_tools/conf/mypy.ini:32]:
```suggestion
# Mocked module in cirq/_compat.py
```

**Review by cduck — APPROVED** [review]
LGTM with two minor comments.

File: cirq-core/cirq/_compat.py [hunk 4]
```diff
@@ -621,6 +621,14 @@ def deprecated_submodule(
             )
 
     def wrap(finder: Any) -> Any:
+        # Sphinx looks for non-wrapped MockFinders
+        # so we have to check for them and not wrap them
+        if 'sphinx' in sys.modules:
+            from sphinx.ext.autodoc.mock import MockFinder
+
+            if isinstance(finder, MockFinder):
+                print("Hello")
```
[the remark this answers is not in the document]
Comment (vtomole) [reply cirq-core/cirq/_compat.py]: Removed

File: cirq-core/cirq/_compat_test.py (hunk 2 above)
Comment (vtomole) [reply cirq-core/cirq/_compat_test.py:922]: Asserted.

**Review by cduck — APPROVED** [review]

**Pull request merged**